docs/tech-notes/life_of_a_query.md

```diff
@@ -1049,173 +1155,147 @@ will eventually block for the corresponding Raft command to be applied
-commands](https://github.com/cockroachdb/cockroach/blob/33c18ad1bcdb37ed6ed428b7527148977a8c566a/pkg/storage/replica.go#L2616)
-calling
-[`processRaftCommand`](https://github.com/cockroachdb/cockroach/blob/33c18ad1bcdb37ed6ed428b7527148977a8c566a/pkg/storage/replica.go#L3247)
-for each one. This will in take the serialized `engine.Batch` and call
-[`replica.applyRaftCommand`](https://github.com/cockroachdb/cockroach/blob/33c18ad1bcdb37ed6ed428b7527148977a8c566a/pkg/storage/replica.go#L3467)
-with it. Here the batch is [deserialized and applied to the
-engine](https://github.com/cockroachdb/cockroach/blob/33c18ad1bcdb37ed6ed428b7527148977a8c566a/pkg/storage/replica.go#L3644)
-and, this time, unlike on the proposed side in
-`applyRaftCommandInBatch`, the changes are actually [committed to
-storage](https://github.com/cockroachdb/cockroach/blob/33c18ad1bcdb37ed6ed428b7527148977a8c566a/pkg/storage/replica.go#L3683). The
-command has now been applied (on one particular replica, but keep in
-mind that the process described in this section happens on *every*
-replica).
-
-We've glossed over something in `processRaftCommand` that's important:
-after applying the command, if the current replica is proposer
-(i.e. the lease holder), we need to signal the proposer (which, as we
-saw in the previous section, is blocked in `tryAddWriteCmd`). This
-happens at the [very
-end](https://github.com/cockroachdb/cockroach/blob/33c18ad1bcdb37ed6ed428b7527148977a8c566a/pkg/storage/replica.go#L3513). We've
-now come full circle - the proposer will now be unblocked and receive
-a response on the channel it was waiting on, and it can unwind the
-stack letting its client know that the request is complete. This reply
-can travel through the hierarchy of `Sender`s, back from the lease
-holder node to the SQL gateway node, to a SQL tree of `planNode`s, to
-the SQL Executor, and, through the `pgwire` implementation, to the SQL
-client.
-
+[`handleRaftReadyRaftMuLocked`](https://github.com/cockroachdb/cockroach/blob/6b9168de33628257f3f1fdb17af1f8205ee21d32/pkg/kv/kvserver/replica_raft.go#L495). This
+will [apply committed entries](https://github.com/cockroachdb/cockroach/blob/6b9168de33628257f3f1fdb17af1f8205ee21d32/pkg/kv/kvserver/replica_raft.go#L839)
+by taking in the serialized `storage.Batch` and [applying it to the
+state machine](https://github.com/cockroachdb/cockroach/blob/6e4c3ae05d92f4104b7b9fbb33f9d1cecf055fd3/pkg/kv/kvserver/apply/task.go#L286)
+by [committing the batch to Pebble](https://github.com/cockroachdb/cockroach/blob/6b9168de33628257f3f1fdb17af1f8205ee21d32/pkg/kv/kvserver/replica_application_state_machine.go#L894).
+The command has now been applied (on one particular replica, but keep in mind
+that the process described in this section happens on *every* replica).
+
+We've glossed over something that's important: after applying the command, if
+the current replica is proposer (i.e. the lease holder), we need to signal the
```

Comment:
maybe it's clearer to say "On the replica on which the client is waiting for the response (i.e. typically the lease holder), we need to signal the channel on which the client is waiting (which, as we ..."

Reply:
Good call, updated.